src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WinRTWebSocket.cs

```diff
@@ -515,5 +539,51 @@ private void UpdateStateCloseReceived()
             }
         }
         #endregion
+
+        #region Helpers
+        private static bool InitMessageWebSocketClientCertificateSupported()
+        {
+            return ApiInformation.IsPropertyPresent(
+                "Windows.Networking.Sockets.MessageWebSocketControl",
+                "ClientCertificate");
+        }
+
+        // There are currently only two ways to convert a .NET X509Certificate object into a WinRT Certificate without
```

Comment:
Why is X509Certificate being used instead of the X509Certificate2 object?  This comments applies generally to this file. In reality, they are probably always X509Certificate2 objects.